src/components/birth-date-picker/birth-date-picker.tsx

```diff
@@ -0,0 +1,115 @@
+// TODO: use MobX.
+// TODO: filter input?
+// TODO: should we update the state when we receive a different value?
+// TODO: date validation belongs in a separate library.
+
+import React = require("react");
+
+export interface BirthDatePickerProps {
+    value?: Date;
+    minDate?: Date;
+    maxDate?: Date;
+    onChange?: (newValue?: Date) => void;
+}
+
+export interface BirthDatePickerState {
+    valid: boolean;
+    year: string;
+    month: string;
+    day: string;
+}
+
+function isDateValid(date: Date): boolean {
+    return !Number.isNaN(date.getTime());
+}
+
+export function dateFromYearMonthDay(y: string, m: string, d: string): Date | undefined {
```

Comment:
I suggest having return type `Date | Error`. I tend to understand `undefined` as "nothing to see here" or "nothing was returned".